### PR TITLE
PULP Event Load Extension Instructions + GNU Tests Version 1.0.0

### PR DESCRIPTION
**Pull request by NandniJamnadas**

### CODE REVIEW

**Review by jeremybennett — CHANGES_REQUESTED** [review]
Overall, looks good. A couple of changes

1. You are lacking a GNU compliant commit message for the documentation commit.
2. You need some assembler tests of the validity of the extension.
    - is ELW only enabled when `-march=...Xcorevelw`
 3. You should include your regression test results as a comment on the PR.

File: bfd/elfxx-riscv.c [hunk 1]
```diff
@@ -2622,6 +2625,8 @@ riscv_multi_subset_supports_ext (riscv_parse_subset_t *rps,
       return "xcorevmem' or `xcorev";
     case INSN_CLASS_COREV_BI:
       return "xcorevbi' or `xcorev";
+    case INSN_CLASS_COREV_ELW:
+      return "xcorevelw' or `xcorev";
```
Comment (jeremybennett) [inline bfd/elfxx-riscv.c:2629]: TBD - what does `corev` mean when we have new versions of the tool chain. Suggestion
- `corev1p0`: Version 1 of all 5 CV32E40Pv1 extensions
- `corev2p0`: Version 2 of all 5 CV32E40Pv1 extensions and version ` of all C32E40Pv2 extensions

File: gas/doc/c-riscv.texi [hunk 2]
```diff
@@ -50,15 +50,17 @@ is provided to enable all CORE-V extensions. To enable only a subset of the CORE
...
+general ALU operations: @samp{Xcorevalu} (version 1 for CV32E40Pv1, version 2 for CV32E40Pv2)
 @item
-Post-incrementing and reg-reg load/store: @samp{Xcorevmem}
+Post-incrementing and reg-reg load/store: @samp{Xcorevmem} (version 1 for CV32E40Pv1, version 2 for CV32E40Pv2)
 @item
-immediate branching: @samp{Xcorevbi}
+immediate branching: @samp{Xcorevbi} (version 1 for CV32E40Pv1, version 2 for CV32E40Pv2)
+@item
+event load: @samp{Xcorevelw} (not supported for CV32E40Pv1, version 1 for CV32E40Pv2)
 @end itemize
```
Comment (jeremybennett) [inline gas/doc/c-riscv.texi:64]: Add `corev` as a catchall.

File: gas/testsuite/gas/riscv/cv-elw-fail.s [hunk 3]
```diff
@@ -0,0 +1,8 @@
+target:
+    # Immediate Boundary Tests
+    cv.elw x5,-2049(x6)
+    cv.elw x5,2048(x6)
```
Comment (jeremybennett) [inline gas/testsuite/gas/riscv/cv-elw-fail.s:4]: What about negative offsets?

**Comment by NandniJamnadas** [comment]
1. "You are lacking a GNU compliant commit message for the documentation commit":
Updated the documentation commit message: https://github.com/openhwgroup/corev-binutils-gdb/pull/50/commits/a9405b6ac6ac60f26acbfe3cd280b4e8764ed4bb
2. "You need some assembler tests of the validity of the extension":
 Added xcorev and xcorevelw tests: https://github.com/openhwgroup/corev-binutils-gdb/pull/50/commits/6f184a9a70ec5fecde3adfec79776896fd3dcce2
3. "You should include your regression test results as a comment on the PR".
GNU Assembler Results:
```
| Category             | Baseline | Curr. Week |       Delta |
| :------------------- | -------: | ---------: | ----------: |
| Expected passes      |      785 |        788 |          +3 |
| Unexpected failures  |        - |          - |           - |
| Unexpected successes |        - |          - |           - |
| Expected failures    |       23 |         23 |           - |
| Unresolved testcases |        - |          - |           - |
| Unsupported tests    |        9 |          9 |           - |
```

File: gas/doc/c-riscv.texi (hunk 2 above)
Comment (jeremybennett) [reply gas/doc/c-riscv.texi:64]: All fixed.

**Review by jeremybennett — APPROVED** [review]

**Comment by jeremybennett** [comment]
All looks good. OK to merge.

**Pull request merged**